### DISCUSSION TITLE
Maintenance and Vulnerability alerts on alternating schedule

### DISCUSSION THREAD

**Discussion by scabana** [renovatebot/renovate#25533] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We would love to have a monthly schedule that would look like this: 

`before 6:00 am on the 1st day of the month`: Update all dependencies
`before 6:00 am on the 15th day of the month`: Update all dependencies if a repository contains vulnerability alerts

We also want to be able to retain scheduling per repository as well(with the same kind of flexibility: interleaved schedules). This would mean we can't setup the Renovate run to force schedules. To add to this, Synopsis Detect is used to track vulnerabilities but I don't expect to have an integration with it, I can deal with that. 

### What you tried so far.

I looked into the package rules' available fields to see if I could have a conditional schedule or even a bypass schedule for a specific rule, but it doesn't seem to be the case. If it was the case, I could have provided a list of repositories that contains vulnerabilities. 

_yes, this could be a solution/feature request:_

it would be great to allow something like this: 

```json
{
  "schedules": [
    {
      "schedule": "CRON|LATER",
      "name": "maintenance"
    },
    {
      "schedule": "CRON|LATER",
      "name": "vulnerabilities"
    }
  ]
}
```

This could lead to a `config.js` that looks something like this: 

```javascript
module.exports = {
  packageRules: [
    {
      "appliesToSchedule": "vulnerabilities",
      "matchRepositories": ".",
      "enabled": false
    },
    {
      "appliesToSchedule": "vulnerabilities",
      "matchRepositories": process.env.SOMEENVVAR.split(","),
      "enabled": true
    }
  ]
};
```

Where `SOMEENVVAR` would be computed outside of renovate.

**Comment by viceice** [comment]
vulnerability updates bypass any schedule.

**Reply by scabana** [reply]
First of all, thanks a lot for the quick reply! 

The scenario is based on vulnerabilities, but I'm not asking that the default vulnerability behavior be configurable. I'm more than willing to do extra work around renovate to accommodate for our process. The issue I got right now is that I can't get to a pattern that would allow me to achieve what I need to achieve. I thought about trying to add some login in the `config.js` file, but I either don't have the specific configured schedule per repo to do something intelligent or I would have to force the schedule centrally. I'd rather not modify Renovate to achieve the goal.
